link-move/src/main/java/com/nhl/link/move/runtime/task/delete/DefaultDeleteBuilder.java

```diff
@@ -51,15 +51,19 @@ public DefaultDeleteBuilder(
         this.type = type;
         this.mapperBuilder = mapperBuilder;
 
-        // always add stats listener
-        stageListener(DeleteStatsListener.instance());
+        setupStatsCallbacks();
+    }
+
+    protected void setupStatsCallbacks() {
+        StatsIncrementor incrementor = StatsIncrementor.instance();
+        stage(DeleteStage.EXTRACT_TARGET, incrementor::targetsExtracted);
+        stage(DeleteStage.COMMIT_TARGET, incrementor::targetsCommitted);
     }
 
     @Override
     protected Class<? extends Annotation>[] supportedListenerAnnotations() {
         return new Class[]{
                 AfterTargetsExtracted.class,
-                AfterSourceRowsExtracted.class,
```

Comment:
from what I can see, there are no usages of `AfterSourceRowsExtracted` listeners in delete task. was it added by mistake?

Reply:
Good catch. Indeed, it doesn't seem to be needed there.